### PR TITLE
Merges mechanics with engineer

### PR DESCRIPTION
**Pull request by ThatFiveGuys**
## About the PR 
This pr merges the mechanics job with engineer job, giving engineers 8 job slots and the access of mechanics. This also adds all the apps that mechanics pdas had into the engineer pda and gives them additional file_amount. This pr doesnt effect the mechanics locker, meaning you can still find the mechanics jumpsuits in them.

To avoid merge conflicts, the spawners for mechanics arent swapped with engineer spawners in this pr, and this is planned to be pred after this is merged. 
## Why's this needed? 

Mechanics are a job with a lot of content but suffer from a lack of aim, whereas engineer are a job with a lot of aim but suffer from lack of content. Not to mention these jobs have a lot of negative overlap. Both jobs are required for repairs, however they cant meaningfully occur at the same time, and the act of just adding back machines on its own or repairing the structure doesnt tend to be super interesting on its own. This means that what happens is one job tends to dominate in doing the repairs (mostly engineers) and the other job sticks mostly to the side. Merging these jobs should help a lot with both jobs. 

## Challenges

These are issues I think merging this pr will bring about, however this will also include potiental solutions to these issues. I think this pr would be worth more then these issues, however Im willing to pr many of these solutions before/after this pr goes through if its viewed as needed.

Problem: People will steal materials from engineering storage to use for the mechanics office.
Potiental solution: Making engineers spawn with some base amount of construction materials (Like #9201) would fix this issue.

Problem: The surplus of people with access to the rkit will make it much harder for security to track people who print out batons and other bad stuff.
Potiental solution: Adding some logging information to the rkit thats only visible to security (Such as what fingerprint/glove id uploaded what blueprint) should fix this issue.

Problem: The increase of people who can roll the syndicate scanner will make a lot of the strategies with the item (such as crusher spam) dull.
Potiental solution: The material cost for many of the items involved in these spammed strategies can be upped, making the spammability and impact of these things lessened in a round. 

## Changelog

```changelog
(u)Ikea
(*)The Mechanic job and their traitor items have been merged into the Engineer job, and the Engineer job has been upped to 8 slots. The contents of the mechanics locker remains the same.
```

### CODE REVIEW

**Comment by Tarmunora** [comment]
I see it's time for #1304 to be rehashed. Again.

**Comment by ThatFiveGuys** [comment]
> I see it's time for #1304 to be rehashed. Again.

Just been over a year and a half since #3838 and the server has changed a lot in that time. I think the previous conversations lacked a lot about what could be done to fix the issues that this pr would likely cause as well, and I hope to bring that to the table.

**Comment by NightmareChamillian** [comment]
Unsure on this, I think it has the capacity to be really cool and a net positive change to the game, but I feel like there'd need to be at least 5 test rounds in order to see how this'd affect the game.

**Comment by Momoberry** [comment]
It is time to rocket react.

**Comment by TDHooligan** [comment]
on one hand engineer rarely has enough to do, on the other hand mechlab already is a pain in the ass with only 3 of them trying to  do their own thing. giving 5 more bored people access to gravitons every round could will likely get tiring very quickly

**Comment by ThatFiveGuys** [comment]
I feel like the main reason why mechanics can be a “shitter role” is just because they dont have much else to do besides that, and promoting construction/station repair among them should shift away from that.

**Comment by UrsulaMejor** [comment]
As we've seen with construction game mode messery, the role of "building a thing" requires both structure building and equipment building. While it can be nice to see players cooperate across departments to make this happen, it's more often the case that players don't even start projects to repair/build because they can't know if the person in the other department will help.

I think it's also the case that being able to start on a project solo in confidence will allow people to get more people interested in joining on the construction; in my experience people are more likely to jump in to help on a thing in progress than to take part in cooperatively start a project.

Construction is, after the ten minutes you spend starting the engine, the engineer's primary job, merging all the pieces of the construction under engineer makes sense to me.

File: code/datums/crew_objective.dm [hunk 1]
```diff
@@ -599,8 +599,8 @@ ABSTRACT_TYPE(/datum/objective/crew/miner)
 		return check_result
 
 
-ABSTRACT_TYPE(/datum/objective/crew/mechanic)
-/datum/objective/crew/mechanic/scanned
+ABSTRACT_TYPE(/datum/objective/crew/engineer)
```
Comment (UrsulaMejor) [inline code/datums/crew_objective.dm:602]: Is the abstract type for engineer crew objectives not already defined???

File: code/datums/jobs.dm [hunk 2]
```diff
@@ -772,32 +772,9 @@ ABSTRACT_TYPE(/datum/job/engineering)
...
-/datum/job/engineering/mechanic
-	name = "Mechanic"
-	limit = 4
-	wages = PAY_DOCTORATE
-
-	slot_back = list(/obj/item/storage/backpack/engineering)
-	slot_belt = list(/obj/item/storage/belt/utility/prepared)
-	slot_jump = list(/obj/item/clothing/under/rank/mechanic)
-	slot_foot = list(/obj/item/clothing/shoes/black)
-	slot_lhan = list(/obj/item/storage/toolbox/electrical/mechanic_spawn)
```
Comment (UrsulaMejor) [inline code/datums/jobs.dm:784]: The engineer starting equipment needs to be adjusted to make sure nothing that mechanics get goes missing in this transfer

**Review by UrsulaMejor — CHANGES_REQUESTED** [review]

File: code/datums/jobs.dm (hunk 2 above)
Comment (ThatFiveGuys) [reply code/datums/jobs.dm:784]: Done, added scanner to engineer

File: code/datums/crew_objective.dm (hunk 1 above)
Comment (ThatFiveGuys) [reply code/datums/crew_objective.dm]: Done!

File: code/datums/jobs.dm (hunk 2 above)
Comment (ThatFiveGuys) [reply code/datums/jobs.dm:784]: Swapped the banana with a scanner

**Comment by UrsulaMejor** [comment]
We're doing it this time. We're making it happen.

**Comment by Sbmhawk** [comment]
The cycle has been broken

**Pull request merged**